roofit/roostats/src/SamplingDistPlot.cxx

```diff
@@ -311,9 +311,9 @@ void SamplingDistPlot::Draw(Option_t * /*options */) {
 
    bool dirStatus = RooPlot::addDirectoryStatus();
    // make the RooPlot managed by this class
-   if (dirStatus) RooPlot::setAddDirectoryStatus(false);
+   RooPlot::setAddDirectoryStatus(false);
    fRooPlot = xaxis.frame();
-   if (dirStatus) RooPlot::setAddDirectoryStatus(true);
+   RooPlot::setAddDirectoryStatus(dirStatus);
```

Comment:
this change does not seem to belong to the commit message, "Make RooPlot react to TH1::AddDirectory()", and I can't see why the new code is equivalent or better than the old code. Naively, I would say you are now never adding `fRooPlot` to `gDirectory`, while before you were in case `dirStatus == true`. Is it desirable to never add this `fRooPlot` to `gDirectory`?

Reply:
Before, if `dirStatus == true`, the global source of dirStatus was set to false, therefore not adding it. If `dirStatus ==false`, also not adding it. I changed this because it's confusing, as you just saw.
I thought that the idiom saveState, flipState, restoreState would be easier to read than having to look up what the functions do only to find out that in all cases it's net being added.

Reply:
I had definitely misread, and the new logic is definitely more readable :+1: